bundle/deploy/metadata/compute.go

```diff
@@ -0,0 +1,59 @@
+package metadata
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/bundle/config/paths"
+	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/bundle/deployment"
+)
+
+type compute struct{}
+
+func Compute() bundle.Mutator {
+	return &compute{}
+}
+
+func (m *compute) Name() string {
+	return "metadata.Compute"
+}
+
+func (m *compute) Apply(_ context.Context, b *bundle.Bundle) error {
+	b.Metadata = deployment.Metadata{
+		Version: deployment.MetadataVersion,
+		Config:  config.Root{},
+	}
+
+	// Set git details in metadata
+	b.Metadata.Config.Bundle.Git = b.Config.Bundle.Git
+
+	// Set job config paths in metadata
+	jobsMetadata := make(map[string]*resources.Job)
+	for name, job := range b.Config.Resources.Jobs {
+		// Compute config file path the job is defined in, relative to the bundle
+		// root
+		relativePath, err := filepath.Rel(b.Config.Path, job.ConfigFilePath)
+		if err != nil {
+			return fmt.Errorf("failed to compute relative path for job %s: %w", name, err)
+		}
+		relativePath = filepath.ToSlash(relativePath)
+
+		// Metadata for the job
+		jobsMetadata[name] = &resources.Job{
+			ID: job.ID,
+			Paths: paths.Paths{
+				RelativePath: path.Clean(relativePath),
```

Comment:
Do we need to clean or `ToSlash` above already does it?

Reply:
ToSlash does not clean the path.